src/app/data/oasis-account-names.ts

```diff
@@ -0,0 +1,110 @@
+  },
+  [Network.testnet]: {
+    [Layer.consensus]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/testnet_consensus.json',
+    [Layer.emerald]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/testnet_paratime.json',
+    [Layer.sapphire]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/testnet_paratime.json',
+  },
+}
+
+const getOasisAccountsMetadata = (network: Network, layer: Layer) =>
+  new Promise<AccountData>((resolve, reject) => {
+    const url = dataSources[network][layer]
+    if (!url) {
+      reject('No data available for this layer')
+    } else {
+      axios.get(url).then(response => {
+        if (response.status !== 200) reject("Couldn't load names")
+        if (!response.data) reject("Couldn't load names")
+        // console.log('Response data is', response.data)
+        const map: AccountMap = new Map()
+        const list: AccountEntry[] = []
+        Array.from(response.data).forEach((entry: any) => {
+          const address = entry.Address
+          const metadata: AccountMetadata = {
+            name: entry.Name,
+            description: entry.Description,
+          }
+          // Register the metadata in its native form
+          list.push({
+            address,
+            ...metadata,
+          })
+          map.set(address, metadata)
+        })
+        resolve({
+          map,
+          list,
+        })
+      }, reject)
+    }
+  })
+
+export const useOasisAccountsMetadata = (network: Network, layer: Layer, enabled: boolean) => {
+  return useQuery(['oasisAccounts', network, layer], () => getOasisAccountsMetadata(network, layer), {
+    enabled,
+    staleTime: Infinity,
+  })
+}
+
+export const useOasisAccountMetadata = (
+  network: Network,
+  layer: Layer,
+  address: string,
+  enabled: boolean,
+): AccountMetadataInfo => {
+  // This is not a condition that can change while the app is running, so it's OK.
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const { isLoading, error, data: allData } = useOasisAccountsMetadata(network, layer, enabled)
+  if (error) {
+    console.log('Failed to load Oasis account metadata', error)
+  }
+  return {
+    metadata: allData?.map.get(address),
+    loading: isLoading,
+  }
+}
+
+export const useSearchForOasisAccountsByName = (
+  network: Network,
+  layer: Layer,
+  nameFragment: string,
+  enabled: boolean,
+) => {
+  const { isLoading, error, data: allData } = useOasisAccountsMetadata(network, layer, enabled)
+  if (error) {
+    console.log('Failed to load Oasis account metadata', error)
+  }
+
+  const textMatcher =
+    nameFragment && enabled
+      ? (entry: AccountEntry): boolean => {
+          return hasTextMatch(entry.name, [nameFragment])
+        }
+      : () => false
+  return {
+    results: (allData?.list || []).filter(textMatcher).map(entry => ({
```

Comment:
This is returning an empty array while loading and while disabled.
Related https://github.com/oasisprotocol/explorer/pull/1356#discussion_r1547059928

Reply:
Yes it does. And why is it a problem?

Reply:
Removed unnecessary fallback values.